trestle/common/list_utils.py

```diff
@@ -100,6 +100,16 @@ def delete_item_from_list(item_list: List[TG], value: TG2, key: Callable[[TG], T
     return item_list
 
 
+def delete_items_from_list(item_list: List[TG], values: List[TG2], key: Callable[[TG], TG2]) -> List[TG]:
+    """Remove the first matching item if it is present in a list based on the callable key matching the query value."""
+    keys = [key(item) for item in item_list]
+    for value in values:
+        if value in keys:
+            index = keys.index(value)
+            del item_list[index]
+    return item_list or []
+
```

Comment:
This is a good use of 'or' but I think it is not needed here because item_list can be [] but it cannot be None

Reply:
It may be slightly cleaner to build the list of indices and then call delete_list_from_list()

Reply:
```
index_list = [keys.index(value) for value in values if value in keys]
delete_list_from_list(..)
```

Reply:
Got it, already adjusted and deleted the function delete_items_from_list or so to avoid confusion and use delete_list_from_list instead